File: web/src/api/history.ts
import request from './request'
import type { VideoHistoryItem, ArticleHistoryItem, SearchHistoryItem, PaginatedResp } from '@/types'

// 视频观看历史列表
export function getVideoHistory(page: number = 1, page_size: number = 20) {
  return request.get<any, PaginatedResp<VideoHistoryItem>>('/history/video/list', {
    params: { page, page_size },
  })
}

// 记录视频观看进度
export function recordVideoView(
  video_id: number,
  progress_sec: number,
  duration_sec: number,
  device: string = 'web'
) {
  return request.post<any, void>('/history/video/view', {
    video_id,
    progress_sec,
    duration_sec,
    device,
  })
}

// 删除单条视频观看历史
export function deleteVideoHistory(video_id: number) {
  return request.delete<any, void>('/history/video', { data: { video_id } })
}

// 清空视频观看历史
export function clearVideoHistory() {
  return request.post<any, void>('/history/video/clear')
}

// 文章阅读历史列表
export function getArticleHistory(page: number = 1, page_size: number = 20) {
  return request.get<any, PaginatedResp<ArticleHistoryItem>>('/history/article/list', {
    params: { page, page_size },
  })
}

// 删除单条文章阅读历史
export function deleteArticleHistory(article_id: number) {
  return request.delete<any, void>('/history/article', { data: { article_id } })
}

// 搜索历史列表
export function getSearchHistory(limit: number = 20) {
  return request.get<any, SearchHistoryItem[]>('/history/search/list', { params: { limit } })
}

// 保存搜索历史
export function saveSearchHistory(keyword: string) {
  return request.post<any, void>('/history/search', { keyword })
}

// 删除单条搜索历史
export function deleteSearchHistory(keyword: string) {
  return request.delete<any, void>('/history/search', { data: { keyword } })
}

// 清空搜索历史
export function clearSearchHistory() {
  return request.post<any, void>('/history/search/clear')
}

File: server/internal/routers/history.go
package routers

import (
	"fake_tiktok/internal/handler"

	"github.com/gin-gonic/gin"
)

// HistoryRouter 用户历史路由注册器
type HistoryRouter struct{}

// InitHistoryRouter 注册历史相关路由
//
// 路由分组（全部私有组，需登录）：
//   - 视频观看历史：
//     POST   /history/video/view     记录观看进度
//     GET    /history/video/list     观看历史列表
//     DELETE /history/video          删除单条
//     POST   /history/video/clear    清空
//   - 文章阅读历史：
//     GET    /history/article/list   阅读历史列表
//     DELETE /history/article        删除单条
//   - 搜索历史：
//     POST   /history/search         保存搜索
//     GET    /history/search/list     搜索历史列表
//     DELETE /history/search         删除单条
//     POST   /history/search/clear   清空
func (r *HistoryRouter) InitHistoryRouter(privateGroup *gin.RouterGroup, handlerGroup *handler.HandlerGroup) {
	history := privateGroup.Group("history")
	{
		// 视频观看历史
		video := history.Group("video")
		{
			video.POST("view", handlerGroup.HistoryHandler.RecordVideoView)
			video.GET("list", handlerGroup.HistoryHandler.ListVideoHistory)
			video.DELETE("", handlerGroup.HistoryHandler.DeleteVideoHistory)
			video.POST("clear", handlerGroup.HistoryHandler.ClearVideoHistory)
		}

		// 文章阅读历史
		article := history.Group("article")
		{
			article.GET("list", handlerGroup.HistoryHandler.ListArticleHistory)
			article.DELETE("", handlerGroup.HistoryHandler.DeleteArticleHistory)
		}

		// 搜索历史
		search := history.Group("search")
		{
			search.POST("", handlerGroup.HistoryHandler.SaveSearchHistory)
			search.GET("list", handlerGroup.HistoryHandler.ListSearchHistory)
			search.DELETE("", handlerGroup.HistoryHandler.DeleteSearchHistory)
			search.POST("clear", handlerGroup.HistoryHandler.ClearSearchHistory)
		}
	}
}
